doc/before-after-reboot-checks.md

```diff
@@ -0,0 +1,48 @@
+# Before and After Reboot Checks
+
+CLUO provides optional functionality for running arbitrary actions before and
+after the rebooting of a node using node labels and annotations.
+
+## Configuring the `update-operator`
+
+When depolying the `update-operator`, you can provide a comma-separated list of
+node annotations to look for before a reboot should occur, and another one for
+after a reboot has occured. The list is provided as a command line argument to
+the binary -
+
+```bash
+update-operator --before-reboot-annotations anno1,anno2 --after-reboot-annotations anno3,anno4
+```
+
+## Making a check
+
+A check that needs to run before reboot should look for the
+`container-linux-update.v1.coreos.com/before-reboot=true` label and a check that
+needs to run after reboot should look for the
+`container-linux-update.v1.coreos.com/after-reboot=true` label. Once it finds
+it's respective label, the check performs whatever function it needs to, and
+then should set one of the annotations provided to the `update-operator` to
+true. 
+
+The actual implementation details of a check are flexible. Provided with this
+repository are examples of a before and an after reboot check, in
+`examples/before-reboot-check.yaml` and `examples/after-reboot-check.yaml`
+respectively. These are implemented as DaemonSet pods which node select on the
+`before-reboot` and `after-reboot` label, and then annotate the node with a
+configured annotation.
+
+## `update-operator` behavior
+
+When a node requests a reboot, the `update-operator` will label the node with
```

Comment:
## Before and After Reboot Labels

`update-operator` labels nodes that are about to reboot `container-linux-update.v1.coreos.com/before-reboot=true` and labels nodes that have just completed rebooting (but not yet scheduable) with `container-linux-update.v1.coreos.com/after-reboot=true`. If you've required before or after reboot annotations, `update-operator` will wait until all annotations are applied before proceeding.